src/rubric_eval/main.py

```diff
@@ -0,0 +1,217 @@
+"""
+How to run:
+
+1. git clone https://github.com/stella-z3g/RubricEval -b public_release
+2. cd RubricEval/ && pip install -e .
+3. python scripts/convert_dataset_to_rubriceval_input_format.py  # This will generate an example instructions.json
+4. rubric_eval get_rubrics --input_path=instructions.json --output_path=instructions_with_rubrics.json
+5. rubric_eval get_completions --model_configs="['gpt-4o-2024-05-13']" --input_path=instructions_with_rubrics.json --output_path_prefix=output_
+6. rubric_eval --input_path_prefix=output_ --output_path_prefix=eval_output_
+"""
+
+
+import sys
+import os
+import ast
+from pathlib import Path
+import pandas as pd
+from typing import List
+from alpaca_eval import utils as ae_utils
+
+import fire
+
+from .helper import get_detailed_rubrics, get_model_completions, get_evaluations
+from .annotators import RubricBrainstormer
+
+CUR_DIR = Path(__file__).parent
+
+__all__ = ["get_rubrics", "get_outputs", "eval_outputs"]
+
+
+def check_df_fields(df, required_fields, optional_fields=set()):
+    allowed_fields = required_fields | optional_fields
+    
+    actual_fields = set(df.columns)
+    
+    # Check if all required fields are present
+    if not required_fields.issubset(actual_fields):
+        missing = required_fields - actual_fields
+        raise RuntimeError(f"Missing required fields: {missing}")
+    
+    # Check if there are any fields not in the allowed set
+    extra_fields = actual_fields - allowed_fields
+    if extra_fields:
+        raise RuntimeError(f"Extra fields found: {extra_fields}")
+    
+
+def cleanup_df_rubrics(df_rubrics):
+    # Drop rows where the "scoring_scales" column is not a dict (because sometimes rubric generator will refuse to generate rubric)
+    mask = df_rubrics["scoring_scales"].apply(lambda x: isinstance(x, dict))
+    return df_rubrics[mask]
+
+
+def list_files_with_prefix(path_prefix, suffix):
```

Comment:
use `from pathlib import Path` it will be much more readable and is the preferred way in newer python version (eg `root / file` instead of `os.path.join(root, file)`). Instead of the for loops you can use `list(Path(...).glob(...))`